docs/contributing.rst

```diff
@@ -62,6 +65,36 @@ can be used to test built assets:
 .. _Wyrm: http://www.github.com/snide/wyrm/
 .. _Sphinx: http://www.sphinx-doc.org/en/stable/
 
+
+_dockerized-build::
+
+Dockerized development
+======================
+
+If you have Docker available on your platform, you can get started building CSS and JS artifacts a bit faster and won't have to worry about any of the setup spilling over into your general environment.
+
+When building with Docker, we create an image containing the build dependencies. Some of these are quite outdated and therefore ideal to isolate a container. The image is tagged as ``sphinx_rtd_theme:latest``.
+
+Inside the running docker image, we mount the working copy of the repository, build the artifacts and finally observe that the artifacts have been built and left in your current git checkout.
+
+Use the following steps:
+
+.. code-block:: console
+
+    # Builds an updated version of the docker image
+    $ docker-compose build
+
+    # Runs the development webserver
+    $ docker-compose run sphinx_rtd_theme dev
+    
+    # If you want to copy stuff out of the Docker environment, run this make
+    # target or read the actual Makefile to see what is going on.
+    # We suggest running this command every time that you want to quickly build
+    # new CSS/JS assets
+    $ make docker-build-all
```

Comment:
We are using `invoke` to handle this type of extra commands in other repositories. I'm not opposite to use Makefile, tho, just saying 😄

Reply:
I'd also prefer to have a singular pattern to use, as much as I like make.

Reply:
Containers provide a quick way to get started. `make` exists on most systems and doesn't sit on top of a large stack that's customized by the project.

With `invoke`, we'd have to firstly bootstrap a Python environment with the correct version of `invoke` - or as a new contributor understand what `invoke` even is. I think there's a better chance that people can copy-paste `make` commands without worrying about setting up an environment for `make`.

If the feeling is "let's not have Make", then I would rather have a shell script for running the `docker-build-all` target, since that's the only command that chains a few things together and should be scripted.

Reply:
I'd still agree that we should stick with a single pattern and use Invoke here. Both Python and Invoke are a normal part of our development workflow and already in use in other repositories, so are reasonable requirements for contributors. The goal is not to make this repository universally accessible as possible, it's just to make a repeatable environment that is easy for the core team to maintain.